airbyte-workers/src/main/java/io/airbyte/workers/DefaultReplicationWorker.java

```diff
@@ -300,6 +300,13 @@ private static Runnable getReplicationRunnable(final AirbyteSource source,
             if (recordsRead % 1000 == 0) {
               LOGGER.info("Records read: {} ({})", recordsRead, FileUtils.byteCountToDisplaySize(messageTracker.getTotalBytesEmitted()));
             }
+          } else {
+            LOGGER.info("Source has no more messages, closing connection.");
+            try {
+              source.close();
+            } catch (final Exception e) {
+              throw new SourceException("Source cannot be stopped!", e);
```

Comment:
I love this exception.  it sounds like Godzilla is coming from the water and cannot be contained